sklearn/discriminant_analysis.py

```diff
@@ -110,11 +112,17 @@ def _class_means(X, y):
     means : array-like of shape (n_classes, n_features)
         Class means.
     """
-    classes, y = np.unique(y, return_inverse=True)
-    cnt = np.bincount(y)
-    means = np.zeros(shape=(len(classes), X.shape[1]))
-    np.add.at(means, y, X)
-    means /= cnt[:, None]
+    xp, is_array_api = get_namespace(X)
+    classes, y = xp.unique_inverse(y)
+    means = xp.zeros(shape=(classes.shape[0], X.shape[1]))
+
+    if is_array_api:
+        for i in range(classes.shape[0]):
+            means[i, :] = xp.mean(X[y == i], axis=0)
+    else:
+        cnt = np.bincount(y)
+        np.add.at(means, y, X)
+        means /= cnt[:, None]
```

Comment:
same here. From the first glance, seems like a much less efficient solution.

Reply:
@thomasjpfan assuming you do not want to change this as part of this PR to keep it focused, please add a TODO comment to re-explore the choice of using the `np.bincount` / `np.add.at` in the numpy branch that seems sub-optimal from a performance point of view.